spring-integration-file/src/main/java/org/springframework/integration/file/splitter/FileSplitter.java

```diff
@@ -199,11 +206,23 @@ public Object next() {
 					return new FileMarker(filePath, Mark.END);
 				}
 				try {
-					return bufferedReader.readLine();
+					if (this.line == null && !this.done) {
+						this.line = bufferedReader.readLine();
+					}
```

Comment:
Yes, it was a hack for "in case" `hasNext()` is not called but you are right, it is incomplete.

I suppose we could change it to...

					if (this.line == null && !this.done) {
						hasNext();
					}

and move the `if (this.eof) ...` below.

WDYT?

I was really disappointed to find BR has no `isOpen()` because `ready()` is not reliable to detect EOF.